### ISSUE TITLE
Compilation fails on '::getdents64'

### ISSUE THREAD

**Issue by rdolbeau** [UoB-HPC/SimEng#203]
Trying to compile on Debian 10 ('buster'), the compilation fails with:

```
/home/dolbeau/SimEng/src/lib/kernel/Linux.cc: In member function 'int64_t simeng::kernel::Linux::getdents64(int64_t, void*, uint64_t)':
/home/dolbeau/SimEng/src/lib/kernel/Linux.cc:525:12: error: '::getdents64' has not been declared
   return ::getdents64(hfd, buf, count);
            ^~~~~~~~~~
/home/dolbeau/SimEng/src/lib/kernel/Linux.cc:525:12: note: suggested alternative: 'SYS_getdents
   return ::getdents64(hfd, buf, count);
            ^~~~~~~~~~
            SYS_getdents64
```
I was able to successfully build SimEng on the same system a couple months ago.

Cordially,

**Comment by jj16791** [comment]
Thanks for posting this issue! We are currently trying to resolve this unexpected error by implementing a custom version of the `getdents64` syscall. After an investigation, multiple host environments seem to be having issues with its calling. We hope to get it done for our `0.9.1` release at the end of next week along with a set of other new features/extensions.

**Comment by rdolbeau** [comment]
I still have the compilation issue; I think it is because `return ::getdents64(hfd, buf, count)` expects to be linked against a glibc wrapper, but that wrapper is not available prior to glibc 2.30. Debian Buster is using 2.28.
Perhaps the code should use an explicit syscall, something like `return syscall(SYS_getdents64, hfd, buf, count)` ?

**Comment by FinnWilkinson** [comment]
Thanks for bringing this up, we are looking into a fix now.
